src/compute/src/render/join/delta_join.rs

```diff
@@ -269,8 +269,8 @@ where
                             update_stream.flat_map_fallible("DeltaJoinFinalization", {
                                 // Reuseable allocation for unpacking.
                                 let mut datums = DatumVec::new();
-                                let mut row_builder = Row::default();
                                 move |row| {
+                                    let mut row_builder = SharedRow::get();
                                     let temp_storage = RowArena::new();
                                     let mut datums_local = datums.borrow_with(&row);
                                     // TODO(mcsherry): re-use `row` allocation.
```

Comment:
Yes, because it refers to the row we get in the input. Instead of dropping it we could do something else with it, although it's not clear how we'd recycle it. I think in the long run, we want to be more specific on how to read data out of the update stream, which might entirely avoid the row allocation.